src/ansys/fluent/core/session.py

```diff
@@ -343,8 +346,87 @@ def download(self, file_name: str, local_file_name: Optional[str] = None):
         FileNotFoundError
             If the remote file does not exist.
         """
-        if self.file_service:
-            if self.file_service.file_exist(file_name):
-                self.file_service.download_file(file_name, local_file_name)
-            else:
-                raise FileNotFoundError("Remote file does not exist.")
+        self._upload_download_helper(
+            is_upload=False, file_name=file_name, local_file_name=local_file_name
+        )
+
+    def _wait_for_file(self, file_name: str):
+        """Wait for file to get ready for upload or download.
+
+        Parameters
+        ----------
+        file_name : str
+            File name
+        Raises
+        ------
+        FileNotFoundError
+            If a file does not exist.
+        """
+        start_time = time.time()
+        max_wait_time = sys.maxsize
+        while (time.time() - start_time) < max_wait_time:
+            if self.file_service.file_exist(os.path.basename(file_name)):
+                break
+            max_wait_time -= 1
+            time.sleep(3)
+        else:
+            raise FileNotFoundError(f"{file_name} does not exist.")
+
+    def _pypim_upload(self, file_name: str, on_uploaded: Optional[Any] = None):
+        """Uploads a file if not available on the server.
+
+        Parameters
+        ----------
+        file_name : str
+            File name
+        on_uploaded: Callable[str]
+            Read a file.
+        Raises
+        ------
+        FileNotFoundError
+            If a file does not exist.
+        """
+        if os.path.isfile(file_name):
+            if not self.file_service.file_exist(os.path.basename(file_name)):
+                self.upload(file_name)
+                self._wait_for_file(file_name)
```

Comment:
I'm wondering why the `_wait_for_file` call is required. Is the `upload` function asynchronous, i.e. it immediately returns before the file-upload is completed?

Reply:
@mkundu1 @seanpearsonuk  Removed `_wait_for_file`. Thank you.

I tested both scenario with and without `_wait_for_file`, both are working fine. Last time we have seen file does not exist error raised by `simple_upload_server` therefore we had added this code block, but it has been removed now.